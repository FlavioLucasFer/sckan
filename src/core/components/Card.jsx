import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';

Card.propTypes = {
	title: PropTypes.string.isRequired,
	className: PropTypes.string,
	cardClassName: PropTypes.string,
	cardTitleClassName: PropTypes.string,
	backgroundColor: PropTypes.string,
	textColor: PropTypes.string,
	actions: PropTypes.arrayOf(Object),
	children: PropTypes.node,
};

function Card(props, ref) {
	const { 
		title, 
		className,
		cardClassName,
		cardTitleClassName,
		backgroundColor, 
		textColor,
		actions,
		children,
		...rest 
	} = props;

	function renderActions() {
		if (!actions) {
			return null;
		}

		return (
			<div className="card-action">
				{actions.map((e, i) => (
					<a key={i} href={e.href}>{e.title}</a>
				))}
			</div>
		);
	}

	return (
		<div ref={ref} {...rest} className={`col ${className || ''}`}>
			<div className={`card ${cardClassName || ''} ${backgroundColor || ''}`}>
				<div className={`card-content  ${textColor || ''}`}>
					<span className={`card-title ${cardTitleClassName || ''}`}>{title}</span>
					<div className="row" 
						style={{
							marginBottom: 0,
						}}>
						{children}
					</div>
				</div>
				{renderActions()}
			</div>
		</div>
	);
}

Card.displayName = 'Card';

export default forwardRef(Card);