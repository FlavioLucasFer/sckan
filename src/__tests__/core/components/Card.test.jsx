import React from 'react';
import { shallow } from 'enzyme';

import Card from 'core/components/Card';

describe('Testing Card core component', () => {
	it('should render correctly', () => {
		const wrapper = shallow(
			<Card title="test">
				<h1>testing</h1>
			</Card>
		);

		expect(wrapper.debug()).toMatchSnapshot();

		const actions = [
			{ title: 'test1', href: 'test1' },
			{ title: 'test2', href: 'test2' },
			{ title: 'test3', href: 'test3' },
		];

		wrapper.setProps({
			title: 'testing',
			className: 's12',
			cardClassName: 'col s12',
			cardTitleClassName: 'bold-text',
			backgroundColor: 'teal',
			textColor: 'grey-text',
			actions,
		});

		expect(wrapper.debug()).toMatchSnapshot();
	});
});