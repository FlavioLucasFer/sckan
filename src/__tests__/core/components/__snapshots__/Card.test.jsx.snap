// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Card core component should render correctly 1`] = `
"<div className=\\"col \\">
  <div className=\\"card  \\">
    <div className=\\"card-content  \\">
      <span className=\\"card-title \\">
        test
      </span>
      <div className=\\"row\\" style={{...}}>
        <h1>
          testing
        </h1>
      </div>
    </div>
  </div>
</div>"
`;

exports[`Testing Card core component should render correctly 2`] = `
"<div className=\\"col s12\\">
  <div className=\\"card col s12 teal\\">
    <div className=\\"card-content  grey-text\\">
      <span className=\\"card-title bold-text\\">
        testing
      </span>
      <div className=\\"row\\" style={{...}}>
        <h1>
          testing
        </h1>
      </div>
    </div>
    <div className=\\"card-action\\">
      <a href=\\"test1\\">
        test1
      </a>
      <a href=\\"test2\\">
        test2
      </a>
      <a href=\\"test3\\">
        test3
      </a>
    </div>
  </div>
</div>"
`;
